loclist.kak

```diff
@@ -0,0 +1,147 @@
+## Location List
+
+declare-option -hidden str loclist_buffer
+declare-option -hidden str-list loclist_stack
+declare-option -docstring %{Defines location patterns for location lists.
+Default locations look like "file:line[:column][:message]"
+
+Capture groups must be:
+    1: filename
+    2: line number
+    3: optional column
+    4: optional message
+} regex loclist_location_format ^\h*\K([^:\n]+):(\d+)\b(?::(\d+)\b)?(?::([^\n]+))
```

Comment:
For me this config option was somewhat underwhelming.
The idea behind it was that it could be used to support different filetypes.
For example, Go, Rust and Python stacktraces all have a different format.
The problem with that was however I tried to set it in "BufSetOption filetype=.." hooks.
Which doesn't work if you invoke :make from different buffers (like the `*make*` buffer itself).

I'm sure we can find a solution in future (tie the option to buffer names?).

(If we keep the regex approach, we can hack Kakoune to support named capture groups though that's probably not super important.)

Reply:
This and `locations_root` are customization points that we may need to see how actually work once we try integrating them with other tools. For now i think its nicer to have it out as an option than hardcoded, even if we don't have a good way to set it